src/crimson/os/seastore/transaction_manager.h

```diff
@@ -491,17 +492,32 @@ class TransactionManager : public SegmentCleaner::ExtentCallbackInterface {
     return segment_cleaner->stat();
   }
 
+  void add_segment_manager(SegmentManager* sm) {
+    LOG_PREFIX(TransactionManager::add_segment_manager);
+    DEBUG("adding segment manager {}", sm->get_device_id());
+    scanner.add_segment_manager(sm);
+    epm->add_allocator(
+      device_type_t::SEGMENTED,
+      std::make_unique<SegmentedAllocator>(
+	*segment_cleaner,
+	*sm,
+	*lba_manager,
+	*journal,
+	*cache));
+  }
+
   ~TransactionManager();
 
 private:
   friend class Transaction;
 
-  SegmentManager &segment_manager;
+  SegmentManager &segment_manager; // primary device
```

Comment:
I don't see an explanation of what "primary device" means.  Does it mean that we only journal to this segment_manager?

Reply:
yes, that's it